### PR TITLE
Lub fix in multigraph qualifier hierarchy

### PR DESCRIPTION
**Pull request by jyluo**

### CODE REVIEW

File: framework/src/org/checkerframework/framework/util/MultiGraphQualifierHierarchy.java [hunk 1]
```diff
@@ -659,6 +665,15 @@ assert getTopAnnotation(a1) == getTopAnnotation(a2) :
         return null;
     }
 
+    // sees if a particular annotation mirror is a polymorphic qualifier
+    private boolean isPolymorphicQualifier(AnnotationMirror qual) {
+        if (AnnotationUtils.containsSame(polyQualifiers.values(), qual)) {
```
Comment (wmdietl) [inline framework/src/org/checkerframework/framework/util/MultiGraphQualifierHierarchy.java]: Just write:
  return AU.containsSame...

It's much shorter and more readable.

File: framework/src/org/checkerframework/framework/util/MultiGraphQualifierHierarchy.java [hunk 2]
```diff
@@ -628,30 +628,36 @@ assert getTopAnnotation(a1) == getTopAnnotation(a2) :
...
+            } else {
+                ErrorReporter.errorAbort("GraphQualifierHierarchy could not determine LUB for " + a1 + " and " + a2 +
+                        ". Please ensure that the checker knows about all type qualifiers.");
             }
         }
         if (outset.size() == 1) {
             return outset.iterator().next();
         }
         if (outset.size() > 1) {
+            // this line is needed by LubGlb checker... the LUB of D & E would otherwise resolve to A even thought it should be C
```
Comment (wmdietl) [inline framework/src/org/checkerframework/framework/util/MultiGraphQualifierHierarchy.java]: Let's discuss this comment in person.

**Comment by wmdietl** [comment]
Is this pull request still current or can it be closed?

**Comment by jyluo** [comment]
The request can be closed, but the content is still current and part of the bigger units request (it's the first commit in that request).

**Pull request closed without merging**